engine/src/cubos/engine/tools/scene_editor/plugin.cpp

```diff
@@ -0,0 +1,268 @@
+static void drawSceneHierarchy(SceneInfo& scene, Commands& cmds, cubos::engine::tools::EntitySelector& selector,
+                               int hierarchyDepth)
+{
+    ImGui::PushID(&scene);
+
+    /// Root node scene
+    bool nodeOpen = ImGui::TreeNode(&scene, "%s", scene.name.c_str());
+    scene.isExpanded = nodeOpen;
+
+    // Remove scene
+    if (hierarchyDepth == 1)
+    {
+        ImGui::SameLine();
+        ImGui::PushID(&scene.shouldBeRemoved);
+        if (ImGui::Button("Remove Scene"))
+        {
+            scene.shouldBeRemoved = true;
+        }
+        ImGui::PopID();
+    }
+
+    if (nodeOpen)
+    {
+        if (hierarchyDepth == 1)
+        {
+            std::string buff = scene.name;
+            ImGui::PushID("nameChange");
+            ImGui::InputText("", &buff, ImGuiInputTextFlags_CallbackCharFilter, EntityNameFilter);
+            ImGui::PopID();
+
+            if (scene.name.compare(buff))
+            {
+                scene.name = buff;
+            }
+        }
+
+        // Add entity to current scene (needs to be root, not a sub scene)
+        if (hierarchyDepth == 0)
+        {
+            if (ImGui::Button("Add Scene"))
+            {
+                std::string sceneName = scene.name + "_scene_" + std::to_string(scene.scenes.size() + 1);
+                SceneInfo newSubscene;
+                newSubscene.name = sceneName;
+                scene.scenes.emplace_back(sceneName, newSubscene);
+            }
+            ImGui::SameLine();
+        }
+        /// Entities
+        drawSceneEntities(scene.entities, scene, selector, cmds, hierarchyDepth);
+
+        /// Scenes
+        std::vector<int> sceneToRemove;
+        for (unsigned int i = 0; i < scene.scenes.size(); i++)
+        {
+            if (scene.scenes[i].second.shouldBeRemoved)
+            {
+                sceneToRemove.push_back(i);
+            }
+            else
+            {
+                drawSceneHierarchy(scene.scenes[i].second, cmds, selector, hierarchyDepth + 1);
+            }
+        }
+        for (const auto& s : sceneToRemove)
+        {
+            scene.scenes.erase(scene.scenes.begin() + s);
+        }
+        ImGui::TreePop();
+    }
+
+    ImGui::PopID();
+}
+
+static void sceneEditorSystem(Commands cmds, Write<SceneInfo> scene,
+                              Write<cubos::engine::tools::EntitySelector> selector)
+{
+    ImGui::Begin("Scene Editor");
+    drawSceneHierarchy(*scene, cmds, *selector, 0);
+    ImGui::End();
+}
+
+void cubos::engine::tools::sceneEditorPlugin(Cubos& cubos)
+{
+    cubos.addPlugin(cubos::engine::imguiPlugin);
+    cubos.addPlugin(cubos::engine::tools::entitySelectorPlugin);
+    cubos.addPlugin(cubos::engine::tools::assetExplorerPlugin);
+
+    cubos.addPlugin(assetsPlugin);
+    cubos.addPlugin(scenePlugin);
+
+    cubos.addResource<SceneInfo>();
+
+    cubos.system(checkAssetEventSystem).tagged("cubos.assets");
```

Comment:
```suggestion
    cubos.system(checkAssetEventSystem);
```

That tag is only important for startup systems, where order really matters.